### PR TITLE
Update build system template arguments for AD number types.

### PR DESCRIPTION
**Pull request by jppelteret**

### CODE REVIEW

File: cmake/config/template-arguments.in [hunk 1]
```diff
@@ -12,15 +12,27 @@ REAL_SCALARS    := { double; float }
...
+// differentiable Sacado scalar types
+DIFFERENTIABLE_TRILINOS_SACADO_REAL_SCALARS := {
+		    @DEAL_II_EXPAND_TRILINOS_SACADO_TYPES@;
+		    }
+
 // differentiable scalar types
 DIFFERENTIABLE_REAL_SCALARS := {
 		    @DEAL_II_EXPAND_TRILINOS_SACADO_TYPES@;
-		    } 
+        @DEAL_II_EXPAND_ADOLC_TYPES@;
```
Comment (davydden) [inline cmake/config/template-arguments.in]: indentation looks off

File: cmake/configure/configure_2_trilinos.cmake [hunk 2]
```diff
@@ -192,7 +192,17 @@ MACRO(FEATURE_TRILINOS_CONFIGURE_EXTERNAL)
   SET(DEAL_II_EXPAND_TRILINOS_BLOCK_SPARSITY_PATTERN "TrilinosWrappers::BlockSparsityPattern")
   SET(DEAL_II_EXPAND_TRILINOS_MPI_BLOCKVECTOR "TrilinosWrappers::MPI::BlockVector")
   SET(DEAL_II_EXPAND_TRILINOS_MPI_VECTOR "TrilinosWrappers::MPI::Vector")
-  SET(DEAL_II_EXPAND_TRILINOS_SACADO_TYPES "Sacado::Fad::DFad<double>; Sacado::Fad::DFad<float>; Sacado::Fad::DFad<Sacado::Fad::DFad<double>>; Sacado::Fad::DFad<Sacado::Fad::DFad<float>>")
+  # Note: Only CMake 3.0 and greater support line continuation with the "\" character
```
Comment (davydden) [inline cmake/configure/configure_2_trilinos.cmake]: i guess we currently support 2.8.2 or something...

**Comment by jppelteret** [comment]
Drats, I forgot that there are no product types defined yet for `Sacado::Rad::ADvar`. I'll fix these issues shortly.

**Comment by Rombur** [comment]
I don't see how the tester failure can be related to this PR. So I'll merge it.

**Comment by tamiko** [comment]
@Rombur @jppelteret  Well, we have a bit of a fallout: http://cdash.kyomu.43-1.org/index.php?project=deal.II

**Comment by Rombur** [comment]
@tamiko should be fixed by #5421 which has just been merged.

**Comment by jppelteret** [comment]
@tamiko What @Rombur said...

**Pull request merged**